grabbit/src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -0,0 +1,56 @@
+package com.twcable.grabbit.jcr
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.apache.jackrabbit.value.DateValue
+
+import javax.annotation.Nonnull
+import javax.jcr.Node as JCRNode
+import javax.jcr.RepositoryException
+
+import static org.apache.jackrabbit.JcrConstants.JCR_LASTMODIFIED
+import static org.apache.jackrabbit.JcrConstants.JCR_PRIMARYTYPE
+
+@CompileStatic
+@Slf4j
+class JCRNodeDecorator {
+
+    @Delegate
+    JCRNode innerNode
+
+
+    JCRNodeDecorator(@Nonnull JCRNode node) {
+        if(!node) throw new IllegalArgumentException("node must not be null!")
```

Comment:
I am trying to understand what will happen incase of the null (I guess edge cases)

Below class and method is indirectly using the `JCRNodeDecorator`. I do not see the exception has been handled anywhere during the call stack.
```<bean id="clientJcrNodesWriter" class="com.twcable.grabbit.client.batch.steps.jcrnodes.JcrNodesWriter" />```
```
    @Override
    void write(List<? extends ProtoNode> nodeProtos) throws Exception {
        Session session = theSession()
        for (ProtoNode nodeProto : nodeProtos) {
            writeToJcr(nodeProto, session)
        }
    }
```
So in this case Spring is executing this method, what will happen to the next node processing? I guess spring should handle the exception but i do not know for certain. Should we handle on our side too incase of exception?